src/webgpu/api/validation/encoding/cmds/render/setPipeline.spec.ts

```diff
@@ -31,4 +52,14 @@ Tests setPipeline should generate an error iff using an 'invalid' pipeline.
 g.test('pipeline,device_mismatch')
   .desc('Tests setPipeline cannot be called with a render pipeline created from another device')
   .paramsSubcasesOnly(kRenderEncodeTypeParams.combine('mismatched', [true, false]))
-  .unimplemented();
+  .fn(async t => {
+    const { encoderType, mismatched } = t.params;
+
+    const pipeline = mismatched
+      ? t.createRenderPipelineForMismatch(t.mismatchedDevice)
+      : t.createRenderPipelineForMismatch(t.device);
```

Comment:
nit: same, createRenderPipelineForMismatch can be inline if it's only used in one place.